src/vmshepherd/presets/preset.py

```diff
@@ -83,36 +83,41 @@ def _reset_counters(self):
         """
         if if_older_than is None or time.time() - self._vms_refresh_time > if_older_than:
             self._vms_refresh_time = time.time()
-            self._vms = await self.iaas.list_vms(self.name)
+            self._runtime_vms = await self.iaas.list_vms(self.name)
+        else:
+            runtime_data = await self.runtime_mgr.get_preset_data(self.name)
+            self._vms = runtime_data.iaas['vms']
 
     async def manage(self):
         """ Manage function docstring"""
-
         await self.refresh_vms()
 
-        vms_stat = Counter([vm.get_state() for vm in self._vms])
-        missing = self.count - len(self._vms) if len(self._vms) < self.count else 0
+        vms_stat = Counter([vm.get_state() for vm in self._runtime_vms])
```

Comment:
`self._runtime_vms` might be undefined in this place, consider `await self.refresh_vms()` went only with `else` branch

Reply:
You are right, the logic was changed.